output/cloud/expv2/output.go

```diff
@@ -72,19 +76,8 @@ func (o *Output) Start() error {
 		bq:          &o.collector.bq,
 	}
 
-	pf, err := output.NewPeriodicFlusher(
-		o.config.MetricPushInterval.TimeDuration(), o.flushMetrics)
-	if err != nil {
-		return err
-	}
-	o.periodicFlusher = pf
-
-	pfc, err := output.NewPeriodicFlusher(
-		o.config.AggregationPeriod.TimeDuration(), o.collectSamples)
-	if err != nil {
-		return err
-	}
-	o.periodicCollector = pfc
+	o.periodicInvoke(o.config.MetricPushInterval.TimeDuration(), o.flushMetrics)
+	o.periodicInvoke(o.config.AggregationPeriod.TimeDuration(), o.collectSamples)
```

Comment:
The unique alternative I have in mind is to have it directly as a function in the Start method.

```go
func (o Out) Start() {
periodic := func() {
  ...
}
periodic(d, collect)
periodic(d, flush)
}
```

Reply:
Can we just modify `PeriodicFlusher` to not call the callback if configured so? We're essentially duplicating what it does to avoid that final call.

Otherwise, if we're fine with the duplication, then this being a method doesn't bother me, since it likely won't be needed anywhere else.

Reply:
We may evaluate extending the current periodic helper but we need to be sure this is the final architecture. As commented here https://github.com/grafana/k6/pull/3082#discussion_r1207875810, I would like to stabilize the rest then go back here and make some attempts for refactoring it in a new PR.